legend-engine-xt-persistence-test-runner/src/main/java/org/finos/legend/engine/testable/persistence/mapper/v2/DatasetMapper.java

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.testable.persistence.mapper.v2;
+
+import org.eclipse.collections.api.RichIterable;
+import org.finos.legend.engine.persistence.components.common.Datasets;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.Dataset;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.DatasetDefinition;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.Field;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.FieldType;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.SchemaDefinition;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.persistence.service.output.ServiceOutputTarget;
+import org.finos.legend.engine.testable.persistence.exception.PersistenceException;
+import org.finos.legend.engine.testable.persistence.mapper.FieldTypeMapper;
+import org.finos.legend.pure.generated.Root_meta_pure_persistence_metamodel_Persistence;
+import org.finos.legend.pure.generated.Root_meta_pure_persistence_metamodel_service_ServiceOutputTarget;
+import org.finos.legend.pure.generated.Root_meta_pure_persistence_metamodel_target_PersistenceTarget;
+import org.finos.legend.pure.generated.Root_meta_pure_persistence_relational_metamodel_RelationalPersistenceTarget;
+import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.Column;
+import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.Database;
+import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.SchemaAccessor;
+import org.finos.legend.pure.m3.coreinstance.meta.relational.metamodel.relation.Table;
+
+import static org.finos.legend.pure.generated.platform_store_relational_functions.Root_meta_relational_metamodel_datatype_dataTypeToSqlText_DataType_1__String_1_;
+
+public class DatasetMapper
+{
+    public static String STAGING_SUFFIX = "_staging";
+    private static String DEFAULT_SCHEMA = "default";
+    private static String H2_PUBLIC_SCHEMA = "PUBLIC";
+
+    public static Dataset getTargetDatasetV2(Root_meta_pure_persistence_metamodel_Persistence persistence) throws Exception
+    {
+        Root_meta_pure_persistence_metamodel_service_ServiceOutputTarget serviceOutputTarget = getTdsServiceOutput(persistence);
```

Comment:
Why we only cover TDS?  Nested flat is leverage by lake use cases, hence not supporting test cases will lead to gaps on testing while confusing users.

Is this something we plan to support?

Reply:
We have scoped support for TDS as the first cut - graph fetch should follow.

Reply:
When we plan to support this?  We have use cases already leveraging GraphFetch, and this will provide inconsistent experience for the users.